apis/v1/dbaas_exoscale_kafka.go

```diff
@@ -0,0 +1,75 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// Remove some fields that are removed by Crossplane anyway.
+// Some properties need to be added and removed by Crossplane (https://doc.crds.dev/github.com/crossplane/crossplane/apiextensions.crossplane.io/CompositeResourceDefinition/v1@v1.10.0)
+//go:generate yq -i e ../generated/appcat.vshn.io_exoscalekafkas.yaml --expression "with(.spec.versions[].schema.openAPIV3Schema.properties; del(.metadata), del(.kind), del(.apiVersion))"
+//go:generate yq -i e ../generated/appcat.vshn.io_exoscalekafkas.yaml --expression "with(.spec.versions[]; .referenceable=true, del(.storage), del(.subresources))"
+
+// Workaround to make nested defaulting work.
+// kubebuilder is unable to set a {} default
+//go:generate yq -i e ../generated/appcat.vshn.io_exoscalekafkas.yaml --expression "with(.spec.versions[]; .schema.openAPIV3Schema.properties.spec.properties.parameters.default={})"
+//go:generate yq -i e ../generated/appcat.vshn.io_exoscalekafkas.yaml --expression "with(.spec.versions[]; .schema.openAPIV3Schema.properties.spec.properties.parameters.properties.maintenance.default={})"
+//go:generate yq -i e ../generated/appcat.vshn.io_exoscalekafkas.yaml --expression "with(.spec.versions[]; .schema.openAPIV3Schema.properties.spec.properties.parameters.properties.service.default={})"
+//go:generate yq -i e ../generated/appcat.vshn.io_exoscalekafkas.yaml --expression "with(.spec.versions[]; .schema.openAPIV3Schema.properties.spec.properties.parameters.properties.size.default={})"
+//go:generate yq -i e ../generated/appcat.vshn.io_exoscalekafkas.yaml --expression "with(.spec.versions[]; .schema.openAPIV3Schema.properties.spec.properties.parameters.properties.network.default={})"
```

Comment:
This is a terrible hack. 

Kubebuilder is unable to set an empty object as a default. However that is needed so that nested defaults can apply. This (sadly abandoned) issue explains this well: https://github.com/kubernetes-sigs/controller-tools/issues/622

We have the same issue for other XRDs and AFAIK there is a ticket to properly fix this. Maybe we're able to fix this in kubebuilder somehow